tests/charms/mongodb_k8s_test_charm/config.yaml

```
@@ -26,3 +26,11 @@ options:
       `{PROVIDED_USER}` should be used only if no value in the `ldap-user-to-do-mapping` config option is provided.
       If this configuration is not provided, a default string will be computed based on the base_dn returned by the GLAuth k8s charm.
       example: “dc=example,dc=com??sub?(&(objectClass=groupOfNames)(member={USER}))"
+  system-users:
+    type: secret
+    description: |
+      Configure the internal system users and their passwords. The passwords will
+      be auto-generated if this option is not set. It is for internal use only
+      and SHOULD NOT be used by applications. This needs to be a Juju Secret URI pointing
```

```suggestion
      and SHOULD NOT be used by applications. This MUST be a Juju Secret URI pointing
```

On top of that, and it goes also with the `structured_config.py` validation, do we know a precise structure for the secret so we can do some validation on it ?

---

it has the  `secret:` prefix but I cannot find the length in the documentation

---

I added a check in `get_secret_from_id` to assert the `secret:` prefix